pkg/pool-weighted/contracts/smart/LiquidityBootstrappingPool.sol

```diff
@@ -0,0 +1,501 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "@balancer-labs/v2-solidity-utils/contracts/openzeppelin/ReentrancyGuard.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/helpers/WordCodec.sol";
+
+import "../BaseWeightedPool.sol";
+
+/**
+ * @dev Weighted Pool with mutable weights, designed to support V2 Liquidity Bootstrapping
+ * Prevents non-owner joins, supports pausing trading (and starting paused), and allows
+ * the owner to schedule weights to change linearly toward ending values between starting and
+ * ending timestamps.
+ */
+contract LiquidityBootstrappingPool is BaseWeightedPool, ReentrancyGuard {
+    using FixedPoint for uint256;
+    using WordCodec for bytes32;
+
+    // Type declarations
+
+    // Use timestamps instead of blocks (used in V1 ConfigurableRightsPool)
+    // End weights are packed as 4 uint64's, like the _normalizedWeights
+    struct GradualUpdateParams {
+        uint256 startTime;
+        uint256 endTime;
+        bytes32 endWeights;
+    }
+
+    // State variables
+
+    // Minimum time over which to compute a gradual weight change (i.e., seconds between timestamps)
+    uint256 public immutable minWeightChangeDuration;
```

Comment:
What's the intended goal of this value?

Reply:
As in V1 - if the controller could change the weights (and therefore prices) instantly, it would require complete trust from users. Longer values = closer to trustless. The users can be sure the weights can change in say, less than several hours.